### ISSUE TITLE
Cannot open RCT scenarios due to "missing objects"

### ISSUE THREAD

**Issue by ShimmerFairy** [OpenRCT2/OpenRCT2#21110]
### Operating System

Gentoo Linux

### OpenRCT2 build

OpenRCT v0.4.6

### Base game

RollerCoaster Tycoon 2

### Area(s) with this issue?

This issue has to do with park objects and/or asset packs

### Describe the issue

When attempting to start a new RCT1 or RCT2 scenario, the park refuses to load due to it missing objects, which is strange to me since I recall copying all the RCT1 and 2 files over way back when I first started playing the game. The objects listed as missing are:

```
openrct2.music.galaxy
openrct2.music.acid
openrct2.music.dodgems
openrct2.music.blizzard
openrct2.music.extraterrestrial
openrct2.music.fairground2
openrct2.music.ragtime2
openrct2.music.prehistoric
```

I'm gathering that this might be OpenRCT2-made versions of various music tracks, but I shouldn't need those to play original RCT scenarios, so I'm puzzled as to why that would stop me from playing them. Curiously, I was able to load an old save file and look around it just fine. I also have a couple other weird asset-related issues concerning the title screen, but this is the only thing that prevents me from actually playing.

### Steps to reproduce

1. Boot up the game
2. Select "start a new game"
3. Choose any scenario
4. Have it fail due to missing objects.

**Comment by SDNick484** [comment]
I am seeing this bug as well, also on Gentoo.  Reverting to 0.4.5 resolves the issue (I had the old ebuild handy).  I see the issue on both RCT1 and RCT2 scenarios (both versions from GOG).

**Comment by ZehMatt** [comment]
Those objects are shipped with OpenRCT2, so copying them to the right folder should solve it.

**Comment by groundflyer** [comment]
Same problem.
The Gentoo ebuild has downloads disabled. Can this be the cause?
```
                -DDISABLE_DISCORD_RPC=ON
                $(usex !dedicated "-DDISABLE_FLAC=$(usex !flac)" "")
                -DDISABLE_GOOGLE_BENCHMARK=ON
                -DDISABLE_GUI=$(usex dedicated)
                -DDISABLE_HTTP=OFF
                -DDISABLE_IPO=ON
                -DDISABLE_NETWORK=OFF
                $(usex !dedicated "-DDISABLE_OPENGL=$(usex !opengl)" "")
                -DDISABLE_TTF=$(usex !truetype)
                $(usex !dedicated "-DDISABLE_VORBIS=$(usex !vorbis)" "")
                -DDOWNLOAD_OBJECTS=OFF
                -DDOWNLOAD_OPENMSX=OFF
                -DDOWNLOAD_OPENSFX=OFF
                -DDOWNLOAD_REPLAYS=OFF
                -DDOWNLOAD_TITLE_SEQUENCES=OFF
                -DENABLE_SCRIPTING=$(usex scripting)
                -DOPENRCT2_USE_CCACHE=OFF
                -DPORTABLE=OFF
                -DSTATIC=OFF
                -DWITH_TESTS=$(usex test)
                -DUSE_MMAP=ON
```

**Comment by ZehMatt** [comment]
> Same problem. The Gentoo ebuild has downloads disabled. Can this be the cause?
> 
> ```
>                 -DDISABLE_DISCORD_RPC=ON
>                 $(usex !dedicated "-DDISABLE_FLAC=$(usex !flac)" "")
>                 -DDISABLE_GOOGLE_BENCHMARK=ON
>                 -DDISABLE_GUI=$(usex dedicated)
>                 -DDISABLE_HTTP=OFF
>                 -DDISABLE_IPO=ON
>                 -DDISABLE_NETWORK=OFF
>                 $(usex !dedicated "-DDISABLE_OPENGL=$(usex !opengl)" "")
>                 -DDISABLE_TTF=$(usex !truetype)
>                 $(usex !dedicated "-DDISABLE_VORBIS=$(usex !vorbis)" "")
>                 -DDOWNLOAD_OBJECTS=OFF
>                 -DDOWNLOAD_OPENMSX=OFF
>                 -DDOWNLOAD_OPENSFX=OFF
>                 -DDOWNLOAD_REPLAYS=OFF
>                 -DDOWNLOAD_TITLE_SEQUENCES=OFF
>                 -DENABLE_SCRIPTING=$(usex scripting)
>                 -DOPENRCT2_USE_CCACHE=OFF
>                 -DPORTABLE=OFF
>                 -DSTATIC=OFF
>                 -DWITH_TESTS=$(usex test)
>                 -DUSE_MMAP=ON
> ```

Most likely, except replays the rest should be fetched.

**Comment by SDNick484** [comment]
So I compared the working 0.4.5 ebuild to the failing 0.4.6 one and the only difference was this patch "openrct2-0.4.4-dont-force-downloads.patch" which enabled OPENMSX and OPENSFX.  I then modified the 0.4.6 ebuild to set those two to "ON" and rebuilt 0.4.6.  This resolved the issue and sound is now working.  The one item to note is Portage doesn't like when packages download inside the ebuild (which is likely why it's disabled in the first place - everything should be called in the manifest).  To work around this, I built with FEATURES="-network-sandbox" otherwise the build failed.

To be clear, here's the diff of my new ebuild vs the current 0.4.6:
`119,120c119,120
< 		-DDOWNLOAD_OPENMSX=OFF
< 		-DDOWNLOAD_OPENSFX=OFF
---
> 		-DDOWNLOAD_OPENMSX=ON
> 		-DDOWNLOAD_OPENSFX=ON
`

**Comment by SDNick484** [comment]
I've created a Gentoo Bugzilla entry for this packaging issue which is on distro side: [921108](https://bugs.gentoo.org/921108)he 

@ConiKost - Tagging you as I believe you're the maintainer on the Gentoo side (apologies if I'm wrong).

**Comment by ConiKost** [comment]
> Those objects are shipped with OpenRCT2, so copying them to the right folder should solve it.

Are we talking about https://github.com/OpenRCT2/objects?

> The Gentoo ebuild has downloads disabled. Can this be the cause?

Well, yes and not. Ebuilds are _never_ supposed to download on-the-fly additional stuff, as OpenrRCT2 script does. That's the reason, why it's disabled.

@SDNick484 Do you want to say, that the enabling of DOWNLOAD_OPENMSX and DOWNLOAD_OPENSFX fixes it?

**Comment by SDNick484** [comment]
Yes, enabling those two options resolves the issue.

**Comment by ConiKost** [comment]
@janisozaur Could you please clarify for me, why scenarios are apparently depend on openmsx/opensfx? Currently its true, that I haven't (yet) packaged them in our Gentoo ebuild, but have plans to do so in future. I am just suprised, that those are needed? What about the original music/sfx?

**Comment by ConiKost** [comment]
@SDNick484 0.4.7 with openmsx/opensfx is now available in Gentoo.

**Comment by SDNick484** [comment]
Thanks, I will test when I get home.

**Comment by SDNick484** [comment]
> @SDNick484 0.4.7 with openmsx/opensfx is now available in Gentoo.

I can confirm the issue is resolved in 0.4.7 ebuild.

**Comment by ConiKost** [comment]
Thanks. I guess we can close then here ;-)
